From: Alexey Dobriyan <adobriyan@gmail.com>
Subject: binfmt: move more stuff undef CONFIG_COREDUMP

struct linux_binfmt::core_dump and struct min_coredump::min_coredump are
used under CONFIG_COREDUMP only.  Shrink those embedded configs a bit.

Link: https://lkml.kernel.org/r/YglbIFyN+OtwVyjW@localhost.localdomain
Signed-off-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/binfmt_elf.c         |    2 ++
 fs/binfmt_elf_fdpic.c   |    2 +-
 fs/binfmt_flat.c        |    2 ++
 include/linux/binfmts.h |    2 ++
 4 files changed, 7 insertions(+), 1 deletion(-)

--- a/fs/binfmt_elf.c~binfmt-move-more-stuff-undef-config_coredump
+++ a/fs/binfmt_elf.c
@@ -101,8 +101,10 @@ static struct linux_binfmt elf_format =
 	.module		= THIS_MODULE,
 	.load_binary	= load_elf_binary,
 	.load_shlib	= load_elf_library,
+#ifdef CONFIG_COREDUMP
 	.core_dump	= elf_core_dump,
 	.min_coredump	= ELF_EXEC_PAGESIZE,
+#endif
 };
 
 #define BAD_ADDR(x) (unlikely((unsigned long)(x) >= TASK_SIZE))
--- a/fs/binfmt_elf_fdpic.c~binfmt-move-more-stuff-undef-config_coredump
+++ a/fs/binfmt_elf_fdpic.c
@@ -83,8 +83,8 @@ static struct linux_binfmt elf_fdpic_for
 	.load_binary	= load_elf_fdpic_binary,
 #ifdef CONFIG_ELF_CORE
 	.core_dump	= elf_fdpic_core_dump,
-#endif
 	.min_coredump	= ELF_EXEC_PAGESIZE,
+#endif
 };
 
 static int __init init_elf_fdpic_binfmt(void)
--- a/fs/binfmt_flat.c~binfmt-move-more-stuff-undef-config_coredump
+++ a/fs/binfmt_flat.c
@@ -102,8 +102,10 @@ static int flat_core_dump(struct coredum
 static struct linux_binfmt flat_format = {
 	.module		= THIS_MODULE,
 	.load_binary	= load_flat_binary,
+#ifdef CONFIG_COREDUMP
 	.core_dump	= flat_core_dump,
 	.min_coredump	= PAGE_SIZE
+#endif
 };
 
 /****************************************************************************/
--- a/include/linux/binfmts.h~binfmt-move-more-stuff-undef-config_coredump
+++ a/include/linux/binfmts.h
@@ -98,8 +98,10 @@ struct linux_binfmt {
 	struct module *module;
 	int (*load_binary)(struct linux_binprm *);
 	int (*load_shlib)(struct file *);
+#ifdef CONFIG_COREDUMP
 	int (*core_dump)(struct coredump_params *cprm);
 	unsigned long min_coredump;	/* minimal dump size */
+#endif
 } __randomize_layout;
 
 extern void __register_binfmt(struct linux_binfmt *fmt, int insert);
_
